lib/Cron/Workers/Bounce.php

```diff
@@ -118,4 +132,23 @@ public function getNextRunDate() {
       ->addMinutes(rand(0, 59))
       ->addSeconds(rand(0, 59));
   }
+
+  private function findPreviousTask(ScheduledTask $task): ?ScheduledTaskEntity {
+    $taskEntity = $this->scheduledTasksRepository->findOneById($task->id);
+    if (!$taskEntity instanceof ScheduledTaskEntity) return null;
+    return $this->scheduledTasksRepository->findPreviousTask($taskEntity);
+  }
+
+  private function saveBouncedStatistics(SubscriberEntity $subscriber, ScheduledTask $task, ?ScheduledTaskEntity $previousTask) {
+    $taskEntity = $this->scheduledTasksRepository->findOneById($task->id);
+    if (!$taskEntity instanceof ScheduledTaskEntity) return null;
```

Comment:
Only a detail here. Here is returned `null`, but the rest of the code returns nothing. You can use a return type declaration `void` and remove `null` here.